### PR TITLE
[parallel] feat: add cpu offload for fsdp2

### PR DESCRIPTION
**Pull request by nono-Sang**
### What does this PR do?

add cpu offload for fsdp2

### Checklist Before Starting

- Search for relative PRs/issues and link here: ...
- PR title follows `[{modules}] {type}: {description}` format (enforced by [check_pr_title.yml](.github/workflows/check_pr_title.yml))
  - **Allowed modules:** `agent`, `ci`, `ckpt`, `config`, `data`, `dist`, `docker`, `docs`, `logging`, `lora`, `misc`, `model`, `omni`, `optim`, `ops`, `parallel`, `perf`, `release`, `task`, `trainer`
  - **Allowed types:** `feat`, `fix`, `refactor`, `chore`, `test`
  - Breaking changes: prepend `[BREAKING]` — e.g. `[BREAKING][parallel, model] feat: dynamic batching`

### Test

> Validation results (training curves, eval metrics) for changes not covered by CI.

### API and Usage Example

> Show API changes and usage examples if applicable.

### Design & Code Changes

> High-level design description and specific change list.

### Checklist Before Submitting

- Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md)
- Applied pre-commit checks
- Added/updated documentation
- If `tasks/` training scripts were moved or renamed: updated `docs/` examples and verified `python3 scripts/ci/check_doc_task_paths.py` passes (also enforced by the **Check doc task paths** CI workflow)
- Added tests to CI workflow (or explained why not feasible)

### CODE REVIEW

File: veomni/distributed/fsdp2/clip_grad_norm.py [hunk 1]
```diff
@@ -133,23 +133,23 @@ def _local_pth_sum(params: List[torch.nn.Parameter], p: float) -> torch.Tensor:
...
 
-    default_device = grads_local[0].device if len(grads_local) > 0 else torch.device(get_device_type())
-    res = torch.tensor(0.0, device=default_device, dtype=torch.float32)
+    reduce_device = torch.device(get_device_type())
+    res = torch.tensor(0.0, device=reduce_device, dtype=torch.float32)
     with torch.no_grad():
         grouped_grads_local = _group_tensors_by_device_and_dtype([grads_local])
         for (device, _), ([device_grads_local], _) in grouped_grads_local.items():
             if _has_foreach_support(device_grads_local, device) or _device_has_foreach_support(device):
                 out = torch._foreach_pow_(torch._foreach_norm(device_grads_local, p), p)
```
[the remark this answers is not in the document]
Comment (nono-Sang) [reply veomni/distributed/fsdp2/clip_grad_norm.py:187]: I don’t think this comment is accurate. `torch._foreach_norm` does return a tuple, but tuple immutability only means the container itself cannot be modified; it does not prevent in-place operations on the Tensor objects stored inside it. `torch._foreach_pow_` applies the in-place update to those tensors and does not raise a TypeError in this case.

**Comment by Luosuu** [comment]
thank you! could you please rebase onto master again?

**Comment by Coach257** [comment]
Please use offload_config https://github.com/ByteDance-Seed/VeOmni/blob/51fc256b17d80c5c6f7f11932d585da7e85c30ad/veomni/arguments/arguments_types.py#L380

**Comment by nono-Sang** [comment]
> thank you! could you please rebase onto master again?

done

**Comment by nono-Sang** [comment]
> Please use offload_config
> 
> https://github.com/ByteDance-Seed/VeOmni/blob/51fc256b17d80c5c6f7f11932d585da7e85c30ad/veomni/arguments/arguments_types.py#L380

This is for activation offload, and this PR is for parameters & gradients & optimizer states offload.

**Review by Luosuu — APPROVED** [review]
